enterprise/frontend/src/embedding-sdk/cli/steps/poll-metabase-instance.ts

```diff
@@ -0,0 +1,58 @@
+import fetch from "node-fetch";
+import ora from "ora";
+
+import type { CliStepMethod } from "embedding-sdk/cli/types/cli";
+
+const delay = (duration: number) =>
+  new Promise(resolve => setTimeout(resolve, duration));
+
+const HEALTH_CHECK_MAX_ATTEMPTS = 60 * 5;
+const HEALTH_CHECK_WAIT = 1000;
+
+export const pollMetabaseInstance: CliStepMethod = async state => {
+  let attempts = 0;
+
+  const spinner = ora(
+    "Waiting for the Metabase instance to be ready (~2 mins)",
+  );
+
+  while (attempts < HEALTH_CHECK_MAX_ATTEMPTS) {
+    // The instance is not yet ready. Show a message so users can anticipate the wait.
+    if (attempts === 1) {
+      spinner.start();
+    }
```

Comment:
I wonder if it'd be easier to understand if we call `spinner.start()` before `while` block since it's unlikely that the health check will return 200 with `attempts === 0` (the first attempt), so we'll probably have to show the spinner anyway.

Reply:
@WiNloSt Nice catch! This was for when the container is already ready to go (therefore we do not need to show the spinner) -- but I think @oisincoveney has updated the PR to prompt the user to delete the entire container, so indeed we do not need the `attempts === 1` check anymore.

Reply:
fixed!